network/gossipNode.go

```diff
@@ -92,7 +92,7 @@ type GossipNode interface {
 
 	// GetHTTPClient returns a http.Client with a suitable for the network Transport
 	// that would also limit the number of outgoing connections.
-	GetHTTPClient(peer HTTPPeer) (*http.Client, error)
+	GetHTTPClient(address string) (*http.Client, error)
```

Comment:
Is there any benefit to consider around using a stronger type than string here? I.E. enforcing formatting etc

Reply:
I added this in previous PR, `HTTPPeer ` type was needed only to get its address to reconstruct p2p transport. both ws and p2p implementations fail on parsing so I think it is OK